python/cudf/cudf/core/column/column.py

```diff
@@ -1948,8 +1948,19 @@ def as_column(
                 raise TypeError(
                     f"Cannot convert a {inferred_dtype} of object type"
                 )
+            elif (
+                cudf.get_option("mode.pandas_compatible")
```

Comment:
I think for this case we want to allow this is `dtype is not None`? e.g.

```python
In [5]: idx_obj = pd.Index([True, False, True], dtype=object)

In [6]: pd.Index(idx_obj, dtype=bool)
Out[6]: Index([True, False, True], dtype='bool')
```

Reply:
Done.